[Core] Remove outdated SDK references

MGaetan89 opened the pull request:
## Proposed Changes

The min SDK version has moved to 23. This PR removes some outdated checks and internal/private support APIs.

## Testing

Test: The `AndroidX Presubmits / build-modules (ubuntu-latest, core)` check on GitHub passes.

MGaetan89 committed:
[Core] Remove outdated SDK references

## Proposed Changes

The min SDK version has moved to 23. This PR removes some outdated checks and internal/private support APIs.

## Testing

Test: The `AndroidX Presubmits / build-modules (ubuntu-latest, core)` check on GitHub passes.

MGaetan89 commented:
@alanv I've updated the PR description. Is it possible to restart the Lint check?

liutikas commented:
As @JakeWharton suggested, please split out deprecation of the APIs into a separate PR. Let's start with just the cleanup

MGaetan89 committed:
Revert deprecating proxy methods

MGaetan89 committed:
Run `./gradlew updateApi`

MGaetan89 commented:
I've reverted the newly deprecated methods and updated the API files.
There are three new Lint warnings. I'll look into them to see if I need to update the code, or just update the baseline file.
```
/home/runner/work/androidx/androidx/core/core/src/main/java/androidx/core/widget/TextViewCompat.java:809: Error: Must be one of: LineBreaker.BREAK_STRATEGY_SIMPLE, LineBreaker.BREAK_STRATEGY_HIGH_QUALITY, LineBreaker.BREAK_STRATEGY_BALANCED [WrongConstant]
        textView.setBreakStrategy(params.getBreakStrategy());
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/androidx/androidx/core/core/src/main/java/androidx/core/widget/TextViewCompat.java:810: Error: Must be one of: Layout.HYPHENATION_FREQUENCY_NORMAL, Layout.HYPHENATION_FREQUENCY_NORMAL_FAST, Layout.HYPHENATION_FREQUENCY_FULL, Layout.HYPHENATION_FREQUENCY_FULL_FAST, Layout.HYPHENATION_FREQUENCY_NONE [WrongConstant]
        textView.setHyphenationFrequency(params.getHyphenationFrequency());
                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   Explanation for issues of type "WrongConstant":
   Ensures that when parameter in a method only allows a specific set of
   constants, calls obey those rules.
/home/runner/work/androidx/androidx/core/core/src/main/java/androidx/core/content/pm/ShortcutManagerCompat.java:890: Error: Method.invoke requires both an upper and lower SDK bounds checks to be safe, and the upper bound must be below SdkVersionInfo.HIGHEST_KNOWN_API. [BanUncheckedReflection from androidx.build]
                            getInstanceMethod.invoke(null, context));
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

MGaetan89 committed:
Run `./gradlew updateLintBaseline`

MGaetan89 committed:
Remove more checks in `NotificationCompat` and `PackageManagerCompat`

MGaetan89 commented:
I've removed a couple more outdated checks and updated the Lint baseline files.
